### DISCUSSION TITLE
关于 Doris 编译踩的一些坑。

### DISCUSSION THREAD

**Discussion by ghost** [apache/doris#28985] [A - General / Q&A]
### 1. 编译fe-core时遇到maven从 cloudera 仓库下载依赖时，报错依赖找不到。
我遇到的是net.sourceforge.czt.dev:java-cup:0.11-a-czt02-cdh依赖找不到，这个是因为cloudera官方修改了 maven 源的地址，可以参考这个 [PR](https://github.com/apache/doris/pull/25348)，本人使用的 2.0.2版本，这个 PR 没有合入到这个版本，所以需要手动更改一下。

### 2. 编译时，报错找不到某 xxx.cpp.a文件，ninja: build stopped: subcommand failed.
这个我印象中遇到过几次，有可能出现的是不同的xxx.cpp.a。这里我感觉像是 thirdparty 中的问题，最开始是直接使用 sh build.sh方式直接编译，极大概率都会遇到这个问题，后面我使用第三方下载仓库手动编译第三方包解决了这个问题。具体可以参考[官方文档](https://doris.apache.org/zh-CN/docs/install/source-install/compilation-general)中的常见问题的第 2 条。

### 3. 编译时，卡在 Build Fronted UI。
这个观察 build.sh 脚本，脚本使用默认源，这里导入一个环境变量，让脚本使用自定义源。
```shell
export CUSTOM_NPM_REGISTRY='https://registry.npm.taobao.org'
```

**Comment by ZuoTiJia** [comment]
第二个问题我也遇到了，貌似是因为最近更新了第三方库，官网的文档中，下载的编译好的third-party的链接不是对应所编译的版本

**Comment by yiguolei** [comment]
@PYKe1 @ZuoTiJia  文档里thirdparty 的链接指向的二进制可能不是最新的。 我一般使用ldb toolchain，直接从源码自己构建thirdparty。 这样所有的编译都是源码级别编译，不会有依赖错误的包的问题。

**Reply by ghost** [reply]
OK，感谢 guolei 大佬回复。

**Reply by tendofuji** [reply]
请问能详细说明一下怎么做不，mac还不会配置ldb

**Comment by fengyunhe** [comment]
Build Frontend UI 开始了，没有给mirror，如果现在中断，再构建前面的还需要再来一遍吗？？？

**Reply by fengyunhe** [reply]
试了，不用~
